addons/stock/models/stock_quant.py

```diff
@@ -1132,6 +1132,41 @@ def _unlink_zero_quants(self):
         quants = self.env['stock.quant'].browse([quant['id'] for quant in self.env.cr.dictfetchall()])
         quants.sudo().unlink()
 
+    @api.model
+    def _clean_reservations(self):
+        reserved_quants = self.env['stock.quant']._read_group(
+            [('reserved_quantity', '!=', 0)],
+            ['product_id', 'location_id', 'lot_id', 'package_id', 'owner_id'],
+            ['reserved_quantity:sum', 'id:recordset'],
+        )
+        reserved_move_lines = self.env['stock.move.line']._read_group(
+            [
+                ('state', 'in', ['assigned', 'partially_available', 'waiting', 'confirmed']),
+                ('quantity', '!=', 0),
+                ('product_id.is_storable', '=', True),
+            ],
+            ['product_id', 'location_id', 'lot_id', 'package_id', 'owner_id'],
+            ['quantity:sum'],
+        )
+        reserved_move_lines = {
+            (product, location, lot, package, owner): reserved_quantity
+            for product, location, lot, package, owner, reserved_quantity in reserved_move_lines
+        }
+        for product, location, lot, package, owner, reserved_quantity, quants in reserved_quants:
+            ml_reserved_qty = reserved_move_lines.get((product, location, lot, package, owner), 0)
+            if location.should_bypass_reservation():
+                quants._update_reserved_quantity(product, location, -reserved_quantity, lot_id=lot, package_id=package, owner_id=owner)
+            elif float_compare(reserved_quantity, ml_reserved_qty, precision_rounding=product.uom_id.rounding) != 0:
```

Comment:
Shouldn't it also have a check if we have more quantity reserved on the move line than the total in the quant?